types/src/block_metadata.rs

```diff
@@ -49,4 +49,8 @@ impl BlockMetadata {
         let vote_maps = lcs::to_bytes(&self.previous_block_votes)?;
         Ok((id, self.timestamp_usecs, vote_maps, self.proposer))
     }
+
+    pub fn get_timestamp_usecs(self) -> u64 {
```

Comment:
is this method necessary? I think we can use `into_inner` and get the timestamp from there?

Reply:
yeah, however that returns a tuple , now we are relying on the order of the tuple.